model/Service/ConcurringSessionService.php

```diff
@@ -67,6 +77,20 @@ public function __construct(
         $this->currentSession = $currentSession ?? PHPSession::singleton();
     }
 
+    public function pauseActiveDeliveryExecution($activeExecution): void
```

Comment:
suggestion: I think this actually pauses _all_ delivery executions for a given user

- `pauseConcurrentSessions()` pauses all executions other than `$activeExecution` in `STATE_ACTIVE` for the current user (sets them to `STATE_PAUSED`)
- `resetDeliveryExecutionState()` pauses `$activeExecution` (calling [StateService::pause()](https://github.com/oat-sa/extension-tao-delivery/blob/master/model/execution/StateService.php#L66) at line 231 of this file).

Then, when `$activeExecution` starts, `DeliveryServer::runDeliveryExecution()` calls ([here](https://github.com/oat-sa/extension-tao-delivery/blob/master/controller/DeliveryServer.php#L268)) [StateService::run()](https://github.com/oat-sa/extension-tao-delivery/blob/master/model/execution/StateService.php#L56), which sets the state back to `STATE_ACTIVE`.

I think it would make sense to rename this to something like `pauseActiveDeliveryExecutionsForUser()` to make it clear that it does not pause just the execution passed as an arg but also all other concurrent executions.